src/App/middlewares/token.js

```diff
@@ -0,0 +1,25 @@
+
+import ErrorResponse from '../utils/errorResponse';
+import findUser from '../../Database/services/user';
+import { verifyToken } from '../helpers/jwt';
+
+export default async (req, res, next) => {
+  try {
+    const token = req.headers.autho.split(' ')[1];
```

Comment:
It doesn't make sense to do the same work twice. You can choose either to have this file or the protect.js file but not both! And the one you choose make sure it is made into a class.